src/MusicPlaylistKernel.java

```diff
@@ -0,0 +1,315 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import javax.sound.midi.Sequence;
+
+import components.sequence.Sequence1L;
+
+/**
+ * A concrete implementation for a Music Playlist component.
+ *
+ * <p>
+ * This component represents a playlist as an ordered sequence of song t tles
+ * (using an OSU {@code Sequence<String>}) plus an integer {@code currentIndex}
+ * that denotes the current song.
+ * </p>
+ *
+ * <p>
+ * <b>Convention (Representation Invariant):</b> l>
+ * <li>If {@code songs.length() == 0}, then {@code currentIndex == 0}</li>
+ * <li>If {@code songs.length() > 0}, then 0 ≤ {@code currentIndex} <
+ * {@code songs.length()}.</li>
+ * </ul>
+ * </p>
+ *
+ * <p>
+ * <b>Correspondence:</b>
+ *
+ * The abstract playlist <s0, s1, …, s(n-1)> with current song sₖ is represented
+ * by: l>
+ * <li>{@code songs.entry(0) == s0, …, songs.entry(n-1) == s(n-1)}</li>
+ * <li>{@code currentIndex == k} (if {@code songs.length() > 0}).</li>
+ * </ul>
+ *
+ * If the playlist is empty, we define the current song to be "No songs in
+ * playlist".
+ * </p>
+ *
+ * <p>
+ * This class implements both the kernel methods (such as {@code addSong},
+ * {@code removeSong}, {@code getCurrentSong}, and {@code nextSo
+ * g}) and the Standard methods (such as {@code cle
+ * r}, {@code newInstance}, {@code transf as well as enhanc d operations (like
+ * {@code previousSong}, {@code shuffle}, {@code contains}, {@code
+ * displayPlaylist}, and {@code length}).
+ * </p>
+ *
+ * @author ...
+ */
+public class MusicPlaylistKernel implements musicPlaylist {
```

Comment:
The naming convention here isn't quite right. I would expect to see `MusicPlaylist1` or `MusicPlaylistOnSequence`. I would also expect the relationship to be `extends` because you should be inheriting from an abstract class.